### PR TITLE
Add all rule's in DiktatRuleSetProvider

### LINKED ISSUE #247: Add all rules in DiktatRuleSetProvider
Check and add all rules in DiktatRuleSetProvider

### PR DESCRIPTION
**Pull request by kentr0w**
### What's done:

Added all rules
closes #247

### CODE REVIEW

**Comment by petertrr** [comment]
The idea was to also add some sort of test to avoid such issues in the future

**Review by petertrr — CHANGES_REQUESTED** [review]
The idea was to also add some sort of test to avoid such issues in the future

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/util/DiktatRuleSetProvider4Test.kt [hunk 1]
```diff
@@ -21,3 +25,22 @@ class DiktatRuleSetProvider4Test(private val ruleSupplier: (rulesConfigList: Lis
...
+    companion object {
+        private val IGNORE_FILE = listOf("DiktatRuleSetProvider")
+    }
+
+    @Test
+    fun `check DiktatRuleSetProviderTest contain all rules`() {
+        val path = "${System.getProperty("user.dir")}/src/main/kotlin/org/cqfn/diktat/ruleset/rules"
+        val rulesName = File(path).walk().filter { it.isFile }.map { it.nameWithoutExtension }.toMutableList()
+        rulesName.removeAll(IGNORE_FILE)
+        val q = DiktatRuleSetProvider().get()
```
Comment (aktsay6) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/util/DiktatRuleSetProvider4Test.kt:40]: Please, rename variable

**Review by aktsay6 — CHANGES_REQUESTED** [review]

**Review by petertrr — CHANGES_REQUESTED** [review]
I'm not exactly positive about retrieving class names from file names and hard-coding directory structure, but as a quick and working as required solution which will affect only our own tests, I think this is fine. I have just a couple of minor style suggestions.

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/util/DiktatRuleSetProvider4Test.kt [hunk 2]
```diff
@@ -21,3 +25,22 @@ class DiktatRuleSetProvider4Test(private val ruleSupplier: (rulesConfigList: Lis
...
+    }
+
+    @Test
+    fun `check DiktatRuleSetProviderTest contain all rules`() {
+        val path = "${System.getProperty("user.dir")}/src/main/kotlin/org/cqfn/diktat/ruleset/rules"
+        val filesName = File(path).walk().filter { it.isFile }.map { it.nameWithoutExtension }.toMutableList()
+        filesName.removeAll(IGNORE_FILE)
+        val rulesName = DiktatRuleSetProvider().get()
+                .map { it.toString().split('.').last() }
+                .map { it.substring(0, it.indexOf('@')) }
```
[the remark this answers is not in the document]
Comment (kentr0w) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/util/DiktatRuleSetProvider4Test.kt:42]: Yes, and it doesn't work

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/util/DiktatRuleSetProvider4Test.kt [hunk 3]
```diff
@@ -36,10 +37,7 @@ class DiktatRuleSetProviderTest {
     fun `check DiktatRuleSetProviderTest contain all rules`() {
         val path = "${System.getProperty("user.dir")}/src/main/kotlin/org/cqfn/diktat/ruleset/rules"
         val filesName = File(path).walk().filter { it.isFile }.map { it.nameWithoutExtension }.filterNot { it in IGNORE_FILE }.toMutableList()
-        val rulesName = DiktatRuleSetProvider().get()
-                .map { it.toString().split('.').last() }
-                .map { it.substring(0, it.indexOf('@')) }
-                .toMutableList()
+        val rulesName = DiktatRuleSetProvider().get().map { it::class.simpleName!! }.toMutableList()
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/util/DiktatRuleSetProvider4Test.kt:40]:
```suggestion
        val rulesName = DiktatRuleSetProvider().get().map { it::class.simpleName!! }
```

**Review by petertrr — APPROVED** [review]

**Review by aktsay6 — APPROVED** [review]
LGTM

**Pull request merged**